src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java

```diff
@@ -274,6 +274,24 @@ public String getDetailsPage(@PathVariable Long submissionId, Model model, final
 		model.addAttribute("analysisType", analysisType);
 		model.addAttribute("mailConfigured", emailController.isMailConfigured());
 
+		/*
+		 * Preview information
+		 */
+		try {
+			if (submission.getAnalysisState()
+					.equals(AnalysisState.COMPLETED)) {
+				if (analysisType.equals(BuiltInAnalysisTypes.PHYLOGENOMICS) || analysisType.equals(BuiltInAnalysisTypes.MLST_MENTALIST)) {
```

Comment:
Wouldn't this be better as a `switch` statement?

Reply:
Was getting a "constant expression required" error since the types are required at compile time for the switch statement to work. So just going to use the if/else